docs/source/dataframe-hive.rst

```diff
@@ -0,0 +1,227 @@
+is already hive-partitioned on the ``'year'`` column.
+
+.. code-block:: python
+
+    >>> dd.read_parquet("output-path", filters=[("year", ">", 2022)])
+
+Writing Parquet Data with Hive Partitioning
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Dask's :func:`to_parquet` function will produce a hive-partitioned
+directory scheme automatically when the ``partition_on`` option is used.
+
+.. code-block:: python
+
+    >>> df.to_parquet("output-path", partition_on=["year", "semester"])
+
+    >>> os.listdir("output-path")
+    ["year=2022", "year=2023"]
+
+    >>> os.listdir("output-path/year=2022")
+    ["semester=fall", "semester=spring"]
+
+    >>> os.listdir("output-path/year=2022/semester=spring")
+    ['part.0.parquet', 'part.1.parquet']
+
+
+It is important to recognize that Dask will **not** aggregate the
+data files written within each of the leaf directories. This is
+because each of the DataFrame partitions is written independently 
+during the execution of the :func:`to_parquet` task graph. In order 
+to write out data for partition `i`, the partition-i write task will
+perform a `groupby` operation on columns ``["year", "semester"]``,
+and then each distinct group will be written to the corresponding
+directory using the file name ``'part.{i}.parquet'``. Therefore, it
+is possible for a hive-partitioned write to produce a large number
+of files in every leaf directory (one for each DataFrame partition).
+
+
+Reading Parquet Data with Hive Partitioning
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+In most cases, :func:`read_parquet` will process hive-partitioned
+data automatically. By default, all hive-partitioned columns will
+be interpreted as categorical columns.
+
+.. code-block:: python
+
+    >>> ddf = dd.read_parquet("output-path", columns=["year", "semester"])
+
+    >>> ddf
+    Dask DataFrame Structure:
+                            year         semester
+    npartitions=4                                  
+                category[known]  category[known]
+                            ...              ...
+                            ...              ...
+                            ...              ...
+                            ...              ...
+    Dask Name: read-parquet, 1 graph layer
+
+    >>> ddf.compute()
+    year semester
+    0  2022     fall
+    1  2022     fall
+    2  2022     fall
+    3  2022   spring
+    4  2022   spring
+    5  2022   spring
+    6  2023     fall
+    7  2023     fall
+
+
+Defining a Custom Partitioning Schema
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+When utilizing ``engine='pyarrow'``, it is possible to specify a custom
+schema for the hive-partitioned columns.
+
+.. code-block:: python
+
+    >>> schema = pa.schema([("year", pa.int16()), ("semester", pa.string())])
+
+    >>> ddf2 = dd.read_parquet(
+    ...     path,
+    ...     columns=["year", "semester"],
+    ...     dataset={"partitioning": {"flavor": "hive", "schema": schema}}
```

Comment:
going into this here is likely a little overkill but just reading this section without much more context, I am surprised to read of a partitioning `flavor`. This immediately spawns questions like "What other flavors are there?", "What is the best flavor for my use case?", etc.
Is this worth going into? Is this documented somewhere?

Reply:
Right - Although we only support the "hive" flavor in `to_parquet`, the pyarrow engine actually allows us to support other flavors in `read_parquet`. The [API docs](https://docs.dask.org/en/stable/generated/dask.dataframe.read_parquet.html) do mention that you can pass in arguments to the `pyarrow.dataset.partitioning` API to define an arbitrary "partitioning". However, I tried to keep this particular doc page focused on "hive" partitioning since that's all we support at write time.